From e1e104c0c3c379d05cea030a8125c825cf3676e5 Mon Sep 17 00:00:00 2001
From: Vladimir Oltean <vladimir.oltean@nxp.com>
Date: Mon, 6 Jan 2020 14:31:42 +0200
Subject: [PATCH 14/26] Revert "net: phylink: call mac_an_restart for
 SGMII/QSGMII inband interfaces too"

This reverts commit 57fe15baac62dad00817e3359cefc123e24f10a2.

Signed-off-by: Vladimir Oltean <vladimir.oltean@nxp.com>
---
 drivers/net/phy/phylink.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/net/phy/phylink.c b/drivers/net/phy/phylink.c
index cf8bc8334400..15137a41ca0e 100644
--- a/drivers/net/phy/phylink.c
+++ b/drivers/net/phy/phylink.c
@@ -356,9 +356,7 @@ static void phylink_mac_config_up(struct phylink *pl,
 static void phylink_mac_an_restart(struct phylink *pl)
 {
 	if (pl->link_config.an_enabled &&
-	    (phy_interface_mode_is_8023z(pl->link_config.interface) ||
-	     pl->link_config.interface == PHY_INTERFACE_MODE_SGMII ||
-	     pl->link_config.interface == PHY_INTERFACE_MODE_QSGMII))
+	    phy_interface_mode_is_8023z(pl->link_config.interface))
 		pl->ops->mac_an_restart(pl->config);
 }
 
-- 
2.17.1

